azurerm/resource_arm_private_dns_cname_record.go

```diff
@@ -0,0 +1,167 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/services/privatedns/mgmt/2018-09-01/privatedns"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmPrivateDnsCNameRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmPrivateDnsCNameRecordCreateUpdate,
+		Read:   resourceArmPrivateDnsCNameRecordRead,
+		Update: resourceArmPrivateDnsCNameRecordCreateUpdate,
+		Delete: resourceArmPrivateDnsCNameRecordDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			// TODO: make this case sensitive once the API's fixed https://github.com/Azure/azure-rest-api-specs/issues/6641
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"zone_name": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
since changing this requires creating a new resource - could we also make this `ForceNew` and add some validation to this? e.g.

```suggestion
				Required: true,
				ForceNew: true,
				ValidateFunc: validation.NoEmptyStrings,
```